.github/workflows/python.yml

```diff
@@ -10,244 +7,57 @@ on:
-  #  macos:
-  #    runs-on: macos-latest
-  #    strategy:
-  #      matrix:
-  #        python-version: [ '3.9', '3.10', '3.11', '3.12' ]
-  #    steps:
-  #      - uses: actions/checkout@v4
-  #        with:
-  #          submodules: true
-  #      - uses: actions/setup-python@v5
-  #        with:
-  #          python-version: ${{ matrix.python-version }}
-  #          architecture: x64
-  #      - uses: dtolnay/rust-toolchain@stable
-  #      - name: Build wheels - x86_64
-  #        uses: PyO3/maturin-action@v1
-  #        with:
-  #          target: x86_64
-  #          args: --release --out dist --sdist -m pywr-python/Cargo.toml
-  #      - name: Install built wheel - x86_64
-  #        run: |
-  #          pip install pytest h5py pandas polars pyarrow click
-  #          pip install pywr --no-index --find-links dist --force-reinstall --no-deps
-  #          python -c "import pywr"
-  #          python -m pytest
-  #      - name: Build wheels - universal2
-  #        uses: PyO3/maturin-action@v1
-  #        with:
-  #          target: universal2-apple-darwin
-  #          args: -i ${{ matrix.python-version }} --release --out dist -m pywr-python/Cargo.toml
-  #      - name: Install built wheel - universal2
-  #        run: |
-  #          pip install pytest h5py pandas polars pyarrow click
-  #          pip install pywr --no-index --find-links dist --force-reinstall --no-deps
-  #          python -c "import pywr"
-  #          python -m pytest
-  #      - name: Upload wheels
-  #        uses: actions/upload-artifact@v4
-  #        with:
-  #          name: wheels
-  #          path: dist
-
   windows:
     runs-on: windows-latest
-    strategy:
-      matrix:
-        # We produce abi3-py39 wheels for Python 3.9 later.
-        # If this is changed, then the PyO3 feature in the pywr-python/Cargo.toml file must also be updated.
-        python-version: [ '3.9' ]
+
     steps:
       - uses: actions/checkout@v4
         with:
           submodules: true
       - uses: swatinem/rust-cache@v2
-      - uses: actions/setup-python@v5
+      - name: Install uv
+        uses: astral-sh/setup-uv@v6
         with:
-          python-version: ${{ matrix.python-version }}
-          architecture: x64
+          enable-cache: true
       - uses: dtolnay/rust-toolchain@stable
-      - name: Build wheels
-        uses: PyO3/maturin-action@v1
-        with:
-          target: ${{ matrix.target }}
-          args: -i ${{ matrix.python-version }} --release --out dist -m pywr-python/Cargo.toml
-      - name: Install built wheel
-        run: |
-          pip install pytest h5py pandas polars pyarrow click
-          pip install pywr --no-index --find-links dist --force-reinstall --no-deps
-          python -c "import pywr"
-          python -m pytest
+      - name: Install the project
+        run: uv sync --locked --all-extras --dev
+      - name: Check formatting
+        run: uv run ruff format --check
```

Comment:
Also check linting - `uv run ruff check .` ?

Reply:
To be more specific, I have some larger updates to the Python extension in a branch that is waiting on #398 to be merged.